worlds/sc2wol/Client.py

```diff
@@ -295,13 +299,25 @@ def on_package(self, cmd: str, args: dict):
                 self.game_speed = 0
             self.all_in_choice = args["slot_data"]["all_in_map"]
             slot_req_table = args["slot_data"]["mission_req"]
+
+            first_item = list(slot_req_table.keys())[0]
             # Maintaining backwards compatibility with older slot data
-            self.mission_req_table = {
-                mission: MissionInfo(
-                    **{field: value for field, value in mission_info.items() if field in MissionInfo._fields}
-                )
-                for mission, mission_info in slot_req_table.items()
-            }
+            if first_item in [str(campaign.id) for campaign in SC2Campaign]:
```

Comment:
Recommendation: put version numbers in the slot data. Ideally every independent component would have an independent version number (client and generation). Makes stuff like this much clearer to read if `generation_version = slot_data.get("campaign_generation_version", (0, 1))`, and this becomes `if generation_version == (0, 1)`.

More extensible in the future, too, for the next time you make breaking changes, and by adding a check if the version is too high now you can also nicely print an error message to the user saying "the slot generated a world that requires a higher client version than you have; please update your .apworld"

Reply:
Added slot data version to slot data: 0bfa79015e7abf146153ddcc725ef1c406ad8f58

Reply:
Looks like you're not pulling in that version / checking it here. Should I make a PR after this one merges to use that check / display an error message to the user?